src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileProviderGlobbingFile.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.Framework.FileSystemGlobbing.Abstractions;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    public class FileProviderGlobbingFile : FileInfoBase
+    {
+        private readonly DirectoryInfoBase _parent;
+        private readonly IFileInfo _fileInfo;
+
+        public FileProviderGlobbingFile(DirectoryInfoBase parent, IFileInfo fileInfo)
+        {
+            _parent = parent;
+            _fileInfo = fileInfo;
+        }
+
+        public override string FullName
```

Comment:
depending on the lifetime of this type and how many times it is called you can either set the values up at the construction time. Or to the csharp6 thing @pranavkm was so excited about:


`public override string FullName => _parent.FullName + Path.DirectorySeparatorChar + Name;`